cubi-tools/prototypes/update_workflow.py

```diff
@@ -0,0 +1,646 @@
+                    f"in branch/version tag '{branch}'."
+                    "\nDo you want to update the workflow version in "
+                    "'pyproject.toml'(y/n)? y"
+                )
+                print(
+                    "Dry run!\n"
+                    "\nMetadata version in 'pyproject.toml' would have been updated "
+                    f"from version '{target_metadata_version}' to version "
+                    f"'{branch_metadata_version}'! and "
+                    "\nWorkflow version in 'pyproject.toml' would have been updated "
+                    f"from version '{target_workflow_version}' to version "
+                    f"'{branch_workflow_version}'!"
+                )
+            else:
+                answer_is_pos = user_response(
+                    "\nYou updated your local repo with the 'template-snakemake' "
+                    f"in branch/version tag '{branch}'."
+                    "\nDo you want to update the workflow version in "
+                    "'pyproject.toml'"
+                )
+
+                if answer_is_pos:
+                    with open(
+                        pathlib.Path(workflow_target, "pyproject.toml"),
+                        "w",
+                        encoding="utf-8",
+                    ) as text_file:
+                        text_file.write(toml.dumps(target_pyproject, encoder=None))
+                    print(
+                        "\nMetadata version in 'pyproject.toml' was updated from "
+                        f"version '{target_metadata_version}' to version "
+                        f"'{branch_metadata_version}'! and "
+                        "\nWorkflow version in 'pyproject.toml' was updated from "
+                        f"version '{target_workflow_version}' to version "
+                        f"'{branch_workflow_version}'!"
+                    )
+                else:
+                    print(
+                        "The 'pyproject.toml' metadata version was NOT updated from "
+                        f"version '{target_metadata_version}' to version "
+                        f"'{branch_metadata_version}' and "
+                        "The 'pyproject.toml' workflow version was NOT updated from "
+                        f"version '{target_workflow_version}' to version "
+                        f"'{branch_workflow_version}'!"
+                    )
+        else:
+            print(
+                "\nMetadata and workflow versions in 'pyproject.toml' are "
+                "up-to-date!\n"
+            )
+    return None
+
+
+def user_response(question, attempt=0):
+    """
+    Function to evaluate the user response to the Yes or No question refarding updating
+    the metadata files.
+    """
+    attempt += 1
+    prompt = f"{question}? (y/n): "
+    answer = input(prompt).strip().lower()
+    pos = ["yes", "y", "yay"]
+    neg = ["no", "n", "nay"]
+    if attempt == 2:
```

Comment:
logically, you should check here first if the user provided a valid y/n answer and not scream at them that they have only one chance left; this should come later before the recursion

Reply:
Done